src/ape/api/networks.py

```diff
@@ -299,18 +299,33 @@ def encode_transaction(
         """
 
     @abstractmethod
-    def decode_logs(self, abi: EventABI, raw_logs: List[Dict]) -> Iterator[ContractLog]:
+    def decode_logs(
+        self, events: Union[EventABI, List[EventABI]], logs: List[Dict]
+    ) -> Iterator["ContractLog"]:
```

Comment:
Wonder if this might be somewhat better UX:
```suggestion
    def decode_logs(
        self, raw_logs: Iterable[Dict], *event_abis: List[EventABI],
    ) -> Iterator["ContractLog"]:
```

Used like this:
```py
txn.decode_logs(project.MyContract.EventA, project.MyContract.EventB, ...)
```
or even like this:
```py
txn.decode_logs(*project.MyContract.events)

Reply:
Your suggested change shows `raw_logs` as the first parameter but the use-cases show the events as coming first. I am guessing the raw data would have to come first in this case since it is `*args` for the event types?

So it would be like:

```python
log_data = get_log_data()  # Pretend
decoded_logs = txn.decode_logs(log_data, project.MyContract.EventA, project.MyContract.EventB)
```

Reply:
So this won't do a ton for UX then...

Inside receipt API implementations we do it like this:


```
self.network.ecosystem.decode_logs(log_filter.events, logs)
```

and you suggestion would make it so we have to do

```
self.network.ecosystem.decode_logs(logs, *log_filter.events)
```

it seems like it add unnecessary work to the Python code and an extra character `*`

Reply:
Yes, but the set of events you can give it can be just one event (no need to wrap in a list), several events of your choosing, or all of the events applied with the `*` args syntax. It feels much more flexible